src/hotspot/share/code/codeHeapState.cpp

```diff
@@ -1384,7 +1384,7 @@ void CodeHeapState::print_usedSpace(outputStream* out, CodeHeap* heap) {
       ast->print_cr("Note: The histogram indicates how much space (as a percentage of all\n"
                     "      occupied space) is used by the blocks in the given size range.\n"
                     "      %ld characters are printed per percentage point.\n", pctFactor/100);
-      ast->print_cr("total size   of all blocks: %7ldM", (total_size<<log2_seg_size)/M);
+      ast->print_cr("total size   of all blocks: " INT64_FORMAT_W(7) "M", (total_size<<log2_seg_size)/M);
```

Comment:
Wrong for 32-bit platforms.

Reply:
I don't quite know what to do here, since on platforms where long < size_t (which in practice is probably just 32 bit or Windows) this division will promote to the type of size_t, which is likely 64 bits in size, but unfortunately %zd cannot be used here since the promotion only happens on said platforms where long < size_t, which will trip a compilation breaking warning on platforms where sizeof long is sizeof size_t, which is probably just 64 bit platforms that are not Windows

Reply:
If we defer the issue of what type `total_size` should be and just keep it as `unsigned long` then the simple and correct fix would be to use `%7lu` would it not?